### DISCUSSION TITLE
Can't figure out how to set user, group, and permissions, for output files.

### DISCUSSION THREAD

**Discussion by Hermit-Style** [seanap/auto-m4b#17] [Q&A]
Noob here with a quick question.
I've got it setup on unraid with docker-compose. It finds my mp3 files and converts them to an m4b, but the new file it outputs has the wrong user, group, and permissions.

I need the user to be "nobody", group to be "users", and permissions 775.

I've tried editing the .yml file under the environment section. 
    environment:
      - PUID=99
      - PGID=100
      - UMASK=002

I've set PUID, PGID, and UMASK, to what I need but it's not applying to the outputted files. Any help would be greatly appreciated. I feel like it's probably something simple I'm missing.

**Comment by seanap** [comment]
When you installed docker did you follow these steps to manage docker as a non-root user?
https://docs.docker.com/engine/install/linux-postinstall/

**Comment by Hermit-Style** [comment]
I'm using docker on unraid, so I didn't think that was needed. I've just installed the docker compose manager plugin. Then copied your yaml file into it.

The container finds my mp3 files and converts them to an m4b. So that part is working. 

But the m4b file that's outputted does not have the owner or permissions I specified in the yaml file.

Do I need an .env file? I have nothing there.

**Reply by seanap** [reply]
Are your errors similar to this thread? https://forums.unraid.net/topic/53533-help-with-file-permissions/

**Comment by Hermit-Style** [comment]
It is similar yes. Any files output from the auto-m4b container get changed to the wrong owner and permissions.

The container templates I use from unraid come setup with PUID:99, PGID:100, UMASK:002.

I don't know how to do that manually or apply it to this situation.

**Reply by seanap** [reply]
See solution to same problem here https://github.com/seanap/auto-m4b/issues/18

**Comment by Hermit-Style** [comment]
I can manually enter chmod and chown after each file conversion but is there a way to automate it?

Changing owner and permission that way does solve my problem but I'm trying to figure out how to automate it inside the container.

**Reply by RunTheGambit** [reply]
I am having the same issue with Docker desktop on Ubuntu, have you had any luck?

**Reply by Hermit-Style** [reply]
I'm still just using chmod and chown on the files after it converts them to m4b. I wasn't able to figure out how to automate that part inside the container.

**Reply by seanap** [reply]
Have you tried to chown and chmod the auto-m4b-tool.sh script file? 

`docker exec -it auto-m4b sh -c 'chown nobody:users -R *'`

**Reply by Hermit-Style** [reply]
Thank you seanap. I'll give that a try this weekend and report back. I was thinking editing the script file was the key to my problem. But I wasn't sure exactly how to word the command. 
For that matter I'll have to figure out how to open and edit the script inside the container. I've never done that but I can probably figure it out. I appreciate you taking the time to reply and help me out.

**Comment by ahillebrandt** [comment]
> Have you tried to chown and chmod the auto-m4b-tool.sh script file?
> 
> `docker exec -it auto-m4b sh -c 'chown nobody:users -R *'`

I took a look at this beause I didn't like this workaround.
The solution is to update the runscript.sh and append an "--gid" to the adduser command:
`    adduser \
        --uid "${PUID}" \
        "${user_name}" \
        --gid "${PGID}"`

Without this specification the User-ID will be taken for the Group-ID which will not work for unraid because it's 99:100.
Hope it helps someone...
